### PR TITLE
add AWS China support

### PR DESCRIPTION
**Pull request by yunjiang29**
Support e2e tests on AWS China regions.

### CODE REVIEW

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh [hunk 1]
```diff
@@ -7,7 +7,10 @@ set -o pipefail
 # TODO: move to image
 curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
 
-export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+case "${CLUSTER_TYPE}" in
+aws-china) export AWS_SHARED_CREDENTIALS_FILE=/var/run/aws-china-credential/.awscred;;
+*) export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred";;
+esac
```
Comment (petr-muller) [inline ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh:13]: This should not be necessary, the secret should come from the cluster profile and there is no reason why it should be in a different path in the aws-chine cluster profile

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-ref.yaml [hunk 2]
```diff
@@ -36,5 +36,9 @@ ref:
     default: "origin-ci-int-aws.dev.rhcloud.com"
     documentation: |-
       A fully-qualified domain or subdomain name. The base domain of the cloud provider is used for setting baseDomain variable of the install configuration of the cluster.
+  credentials:
+  - namespace: test-credentials
+    name: aws-china-credential
+    mount_path: /var/run/aws-china-credential
```
Comment (petr-muller) [inline ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-ref.yaml:42]: This should not be necessary, the secret should come from the cluster profile.

File: ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-ref.yaml [hunk 3]
```diff
@@ -10,3 +10,7 @@ ref:
       memory: 300Mi
   documentation: |-
     The IPI deprovision step tears down the cluster.
+  credentials:
+  - namespace: test-credentials
+    name: aws-china-credential
+    mount_path: /var/run/aws-china-credential
```
Comment (petr-muller) [inline ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-ref.yaml:16]: This should not be necessary, the secret should come from the cluster profile.

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-ref.yaml [hunk 4]
```diff
@@ -17,5 +17,9 @@ ref:
     env: OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
   - name: "release:latest"
     env: RELEASE_IMAGE_LATEST
+  credentials:
+  - namespace: test-credentials
+    name: aws-china-credential
+    mount_path: /var/run/aws-china-credential
```
Comment (petr-muller) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-ref.yaml:23]: This should not be necessary, the secret should come from the cluster profile.

File: ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh [hunk 5]
```diff
@@ -6,7 +6,11 @@ set -o pipefail
 
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
 
-export AWS_SHARED_CREDENTIALS_FILE=$CLUSTER_PROFILE_DIR/.awscred
+case "${CLUSTER_TYPE}" in
+aws-china) export AWS_SHARED_CREDENTIALS_FILE=/var/run/aws-china-credential/.awscred;;
+*) export AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred;;
+esac
+
```
Comment (petr-muller) [inline ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh:13]: This should not be necessary, the secret should come from the cluster profile and there is no reason why it should be in a different path in the aws-chine cluster profile

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 6]
```diff
@@ -61,6 +61,7 @@ fi
 
 case "${CLUSTER_TYPE}" in
 aws|aws-arm64) export AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred;;
+aws-china) export AWS_SHARED_CREDENTIALS_FILE=/var/run/aws-china-credential/.awscred;;
```
Comment (petr-muller) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:64]: This should not be necessary, the secret should come from the cluster profile and there is no reason why it should be in a different path in the aws-chine cluster profile

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh:13]: @petr-muller per your comments in slack,  I've created a new pr for adding new profiles https://github.com/openshift/release/pull/21734

I have an aws-china credential in `test-credentials/aws-china-credential`, which has been added in vault.ci.openshift.org, how they could be added in cluster profile? 
I saw the `.awscred` in aws profile is https://github.com/openshift/release/blob/master/core-services/ci-secret-bootstrap/_config.yaml#L709-L711 , but what's the `field` and `item`?
Is there a sample code that I can reference? thanks.

Comment (petr-muller) [reply ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh:13]: You can sync your `.awscred` file from your vault into the cluster profile secret (the same you configured in the ci-secret-bootstrap as destination) by setting the secretsync keys. I have submitted a docs PR https://github.com/openshift/ci-docs/pull/184 in a hope to clarify this process:

```
secretsync/target-namespace: "ci"
secretsync/target-name: "cluster-secrets-<name>"
```

Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh]: thanks @petr-muller , I will have try.

**Comment by yunjiang29** [comment]
Hello @patrickdillon , I added a new e2e test for AWS China, the installation process looks good, but some post tests failed due to `no such host` error [1], not sure if this is a DNS issue or something incorrect in this PR, could you please take a look at this? Thanks!

[1] https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/21554/rehearse-21554-periodic-ci-openshift-release-master-nightly-4.9-e2e-aws-china/1437679707561660416

**Pull request closed without merging**